odl/solvers/functional/functional.py

```diff
@@ -64,7 +64,12 @@ def __init__(self, space, linear=False, grad_lipschitz=np.nan):
         # Cannot use `super(Functional, self)` here since that breaks
         # subclasses with multiple inheritance (at least those where both
         # parents implement `__init__`, e.g., in `ScalingFunctional`)
-        Operator.__init__(self, domain=space, range=space.field, linear=linear)
+#        print(range)
+        if range is None:
+            range = domain.field
+        elif range != domain.field:
+            linear = False
```

Comment:
After some very recent discussions (see #1331), we decided for practical reasons to go with linearity in the sense "C = R^2". That would mean here that this `elif` case should go away.